<launch>
    <!-- A little different from other launch files: dictates that the robots
    are each in their own namespace, rather than placing the map_merger in its
    own namespace. -->
    <arg name="ns" default="false"/>

    <group if="$(arg ns)">
        <node name="map_merger" pkg="exploration_map" type="map_merger_node">
            <rosparam command="load" file="$(find exploration_map)/config/map_merger/map_merger.yaml"/>
            <rosparam command="load" file="$(find utacc_exploration)/config/exploration_common.yaml"/>
            <remap from="robot_0_map" to="segbot/segbot_map"/>
            <remap from="robot_0_pose" to="segbot/segbot_pose"/>
            <remap from="robot_0_goal" to="segbot/move_base_simple/goal"/>
            <remap from="robot_1_map" to="hexacopter/hexa_map"/>
            <remap from="robot_1_pose" to="hexacopter/hexa_pose"/>
            <remap from="robot_1_goal" to="hexacopter/goal_pose"/>
        </node>
    </group>
    <group unless="$(arg ns)">
        <node name="map_merger" pkg="exploration_map" type="map_merger_node">
            <rosparam command="load" file="$(find exploration_map)/config/map_merger/map_merger.yaml"/>
            <rosparam command="load" file="$(find utacc_exploration)/config/exploration_common.yaml"/>
            <remap from="robot_0_map" to="segbot_map_e2"/>
            <remap from="robot_0_pose" to="segbot_pose_2"/>
            <remap from="robot_0_goal" to="segbot_goal"/>
            <remap from="robot_1_map" to="hexa_map_e2"/>
            <remap from="robot_1_pose" to="hexa_pose_e2"/>
            <remap from="robot_1_goal" to="hexa_goal"/>
        </node>
    </group>
</launch>
